ci-operator/step-registry/cucushift/installer/check/vsphere/cpms/cucushift-installer-check-vsphere-cpms-commands.sh

```diff
@@ -0,0 +1,91 @@
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+
+
+function checkMultizone() {
+    check_result=0
+    cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices')
+    cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
+    cpms_template=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
+#check network,workspace and template field in cpms.
+    if [[ ${cpms_network} == "null" ]] && [[ ${cpms_workspace} == "{}" ]] && [[ ${cpms_template} == "" ]]; then
+        echo "INFO: For multi failure domains The network,workspace and template under cpms are empty, that's expected"
+    else
+	echo "ERROR: For multi failure domains The network,workspace and template under cpms should be empty. network is ${cpms_network},workspace is ${cpms_workspace},template is ${cpms_template}" 
+        check_result=1
+    fi
+    
+#check the name of failuredomains in cpms should be same with infrastructure.
+    fd_name_cpms=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name')
+    fd_name_infra=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].name' | head -n3)
+    if [[ ${fd_name_cpms} == "${fd_name_infra}" ]]; then
+	echo "INFO: The failure domain name are same between infrastructure and cmps"
+    else
+	echo "ERROR: The failure domain name are different between infrastructure and cmps. please check"
+        check_result=1
+    fi
+    return ${check_result}
+}
+
+
+function checkSinglezone() {
+        check_result=0
+	node_name=$(oc get machines.machine.openshift.io -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers | awk '{print $1}' | head -n1)
+	machine_workspace=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.workspace')
+	machine_template=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.template')
+	machine_network=$(oc get machine ${node_name} -n openshift-machine-api -o json | jq -r '.spec.providerSpec.value.network.devices[].networkName')
+        
+	cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -o json | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices[].networkName')
+	cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
```

Comment:
A minor comment about `-o json` v.s `-ojson`, suggest to sync them into one format, so that other can easily search the same command line in the script.

Reply:
ok